roles/translator/src/proxy/bridge.rs

```diff
@@ -95,49 +137,74 @@ impl Bridge {
         let tx_sv2_submit_shares_ext = self_
             .safe_lock(|s| s.tx_sv2_submit_shares_ext.clone())
             .unwrap();
+        let target = self_.safe_lock(|s| s.target.clone()).unwrap();
         task::spawn(async move {
             loop {
-                let (sv1_submit, extranonce) = rx_sv1_submit.clone().recv().await.unwrap();
+                let upstream_target: [u8; 32] =
+                    target.safe_lock(|t| t.clone()).unwrap().try_into().unwrap();
```

Comment:
can this fail?

Reply:
no as soon as we pass a 32 byte vector from upstream.rs